public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts

```diff
@@ -150,32 +150,36 @@ function reRunBuiltInAnnotationsLayer(scene: DashboardScene) {
 }
 
 export function getAdHocFilterSetFor(scene: DashboardScene, ds: DataSourceRef | null | undefined) {
-  const controls = scene.state.controls ?? [];
+  const variables = sceneGraph.getVariables(scene);
 
-  for (const control of controls) {
-    if (control instanceof AdHocFilterSet) {
-      if (control.state.datasource === ds || control.state.datasource?.uid === ds?.uid) {
-        return control;
+  for (const variable of variables.state.variables) {
+    if (sceneUtils.isAdHocVariable(variable)) {
+      const filtersDs = variable.state.datasource;
+      if (filtersDs === ds || filtersDs?.uid === ds?.uid) {
+        return variable;
       }
     }
   }
 
-  const newSet = new AdHocFilterSet({ datasource: ds });
+  const newVariable = new AdHocFiltersVariable({
+    name: 'Filters',
+    datasource: ds,
+  });
 
   // Add it to the scene
-  scene.setState({
-    controls: [controls[0], newSet, ...controls.slice(1)],
+  variables.setState({
+    variables: [...variables.state.variables, newVariable],
   });
 
-  return newSet;
+  return newVariable;
 }
 
-function updateAdHocFilterSet(filterSet: AdHocFilterSet, newFilter: AdHocFilterItem) {
+function updateAdHocFilterSet(filterSetVariable: AdHocFiltersVariable, newFilter: AdHocFilterItem) {
   // Check if we need to update an existing filter
-  for (const filter of filterSet.state.filters) {
+  for (const filter of filterSetVariable.state.filters) {
     if (filter.key === newFilter.key) {
-      filterSet.setState({
-        filters: filterSet.state.filters.map((f) => {
+      filterSetVariable.setState({
+        filters: filterSetVariable.state.filters.map((f) => {
           if (f.key === newFilter.key) {
             return newFilter;
```

Comment:
nitpit: filtersVar instead of filterSetVariable   (it's what I name most instance variables in the scenes lib)